src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -78,20 +89,33 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	// Establish outbound connection.
-	addr := hostIncludingPort(req, 443)
-	d := client.proxiedDialer(func(network, addr string) (net.Conn, error) {
-		// UGLY HACK ALERT! In this case, we know we need to send a CONNECT request
-		// to the chained server. We need to send that request from chained/dialer.go
-		// though because only it knows about the authentication token to use.
-		// We signal it to send the CONNECT here using the network transport argument
-		// that is effectively always "tcp" in the end, but we look for this
-		// special "transport" in the dialer and send a CONNECT request in that
-		// case.
-		return client.getBalancer().Dial("connect", addr)
-	})
-
-	connOut, err = d("tcp", addr)
+	sendToProxy := false
+	for _, proxiedPort := range client.ProxiedCONNECTPorts {
+		if port == proxiedPort {
+			sendToProxy = true
+			break
+		}
+	}
```

Comment:
My one thought here @oxtoacart is testability. This class is one of our worst offenders in terms of test coverage (https://coveralls.io/builds/5259682), partly because handler_test.go I don't think actually tests anything in handler.go! A simple step would be to make this a separate sendToProxy(req *http.Request) method, something like that, which would be very easy to also write a test for. If you wanted to take that further, making handler_test.go really hit ServeHTTP would be pretty sweet, but we could also leave that for now.